### PR TITLE
Spelling fix in documentation.

### PR DESCRIPTION
**Pull request by koraytugay**
Spelling fix in documentation.

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@koraytugay Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-batch&pullRequestId=606)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-batch&pullRequestId=606) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@koraytugay Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-batch&pullRequestId=606)!

File: spring-batch-infrastructure/src/main/java/org/springframework/batch/item/database/JdbcBatchItemWriter.java [hunk 1]
```diff
@@ -41,7 +41,7 @@
  * {@link NamedParameterJdbcTemplate} to execute a batch of statements for all items
  * provided.</p>
  *
- * The user must provide an SQL query and a special callback in the for of either
+ * The user must provide an SQL query and a special callback in the favor of either
```
Comment (Buzzardo) [inline spring-batch-infrastructure/src/main/java/org/springframework/batch/item/database/JdbcBatchItemWriter.java]: It should be "The user must provide an SQL query and a special callback for either of"

It could be "The user must provide an SQL query and a special callback in favor of either"

However, I tend to prefer the shorter wording whenever possible, since no one is reading this stuff for fun.

**Comment by mminella** [comment]
Since the original repo is gone for this PR, I added it myself via 7e402242db42d9b0562fa4135c9093ee9b17f1ca.  Thanks for the contribution!
